api/src/main/java/uk/gov/dvsa/mot/app/MotrRequestHandler.java

```diff
@@ -0,0 +1,199 @@
+package uk.gov.dvsa.mot.app;
+
+import com.amazonaws.serverless.proxy.RequestReader;
+import com.amazonaws.serverless.proxy.model.ApiGatewayRequestContext;
+import com.google.common.base.Strings;
+import com.google.inject.Inject;
+
+import org.apache.log4j.Logger;
+
+import uk.gov.dvsa.mot.trade.api.InternalServerErrorException;
+import uk.gov.dvsa.mot.trade.api.InvalidResourceException;
+import uk.gov.dvsa.mot.trade.api.MotrResponse;
+import uk.gov.dvsa.mot.trade.api.TradeException;
+import uk.gov.dvsa.mot.trade.read.core.MotrReadService;
+import uk.gov.dvsa.mot.trade.service.AnnualTestExpiryDateCalculator;
+import uk.gov.dvsa.mot.vehicle.hgv.HgvVehicleProvider;
+import uk.gov.dvsa.mot.vehicle.hgv.model.HgvPsvVehicle;
+import uk.gov.dvsa.mot.vehicle.hgv.model.TestHistory;
+import uk.gov.dvsa.mot.vehicle.hgv.model.Vehicle;
+
+import java.util.Optional;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.core.Response;
+
+@Path("/")
+public class MotrRequestHandler extends AbstractRequestHandler {
+    private static final Logger logger = Logger.getLogger(MotrRequestHandler.class);
+
+    private MotrReadService motrReadService;
+    private HgvVehicleProvider hgvVehicleProvider;
+    private String awsRequestId;
+
+    public MotrRequestHandler() {
+        super();
+    }
+
+    public MotrRequestHandler(boolean injectSelf) {
```

Comment:
is it useful at all ?

Reply:
From what I see its a workaround for transitive Guice injections for unit tests purposes. I'm guessing the empty constructor is for guice in the live code - can check that later